### PR TITLE
Remove references to v2 commands, remove unused functions, and constants

### LINKED ISSUE #5636: Cleanup references of v2 commands from functions, comments, and templates
There are some references to commands such as `odo config`, `odo create` , `odo watch`, and other v2 commands in the code in utility functions, comments and test templates. Those require cleaning up.

_Originally posted by @valaparthvi in https://github.com/redhat-developer/odo/issues/5521#issuecomment-1090239523_

### PR DESCRIPTION
**Pull request by valaparthvi**
**What type of PR is this:**
/kind code-refactoring

**What does this PR do / why we need it:**

**Which issue(s) this PR fixes:**

Fixes #5636 

**PR acceptance criteria:**

- [ ] Unit test 

- [ ] Integration test 

- [ ] Documentation 

**How to test changes / Special notes to the reviewer:**
1. Used Goland editor inspection tool to identify unused functions, and constants.

### CODE REVIEW

**Comment by feloy** [comment]
As we will rewrite many commands that were existing in v2 and that have been removed for the time being, I would prefer not to remove the utility functions that are not used now but will most probably be helpful when we will write the new commands.

I would prefer to delete these ones when we are done with v3.

**Comment by valaparthvi** [comment]
> As we will rewrite many commands that were existing in v2 and that have been removed for the time being, I would prefer not to remove the utility functions that are not used now but will most probably be helpful when we will write the new commands.
> 
> I would prefer to delete these ones when we are done with v3.

Can you point out the functions? A few at least?

**Comment by feloy** [comment]
> > As we will rewrite many commands that were existing in v2 and that have been removed for the time being, I would prefer not to remove the utility functions that are not used now but will most probably be helpful when we will write the new commands.
> > I would prefer to delete these ones when we are done with v3.
> 
> Can you point out the functions? A few at least?

Here are some functions I think it would be interesting to keep for the moment:

- `GetProjectTypeFromDevfileMetadata`
- `GetLanguageFromDevfileMetadata` to be exhaustive on `Get*FromDevfileMetadata`
- `PreStart DevfileEventType = "preStart"` and other event types to be exhuastive
- all methods in `pkg/log/status.go` to have more choices on displaying output

**Comment by valaparthvi** [comment]
> > > As we will rewrite many commands that were existing in v2 and that have been removed for the time being, I would prefer not to remove the utility functions that are not used now but will most probably be helpful when we will write the new commands.
> > > I would prefer to delete these ones when we are done with v3.
> > 
> > 
> > Can you point out the functions? A few at least?
> 
> Here are some functions I think it would be interesting to keep for the moment:
> 
>     * `GetProjectTypeFromDevfileMetadata`
> 
>     * `GetLanguageFromDevfileMetadata` to be exhaustive on `Get*FromDevfileMetadata`
> 
>     * `PreStart DevfileEventType = "preStart"` and other event types to be exhuastive
> 
>     * all methods in `pkg/log/status.go` to have more choices on displaying output

I removed the Get*FromDevfileMetadata methods because they were added IIRC to add them to telemetry data, I do not believe they would be useful someplace else.

The devfile events have been around for a long time and they haven't been put to use, so I removed them. And they're anyway constants, we can add them back with v2 as a reference.

About methods in `pkg/log/status.go`, they've been around for quite sometime(even during v2)  in an unused state. The new commands have a different UI/UX, I figured we would write new methods as and when required.

**Comment by valaparthvi** [comment]
@feloy Can you lgtm again? I had to rebase.

**Review by feloy — APPROVED** [review]

File: pkg/component/component.go [hunk 1]
```diff
@@ -35,26 +35,8 @@ func GetComponentTypeFromDevfileMetadata(metadata devfile.DevfileMetadata) strin
 }
 
 // GetProjectTypeFromDevfileMetadata returns component type from the devfile metadata
```
Comment (dharmit) [inline pkg/component/component.go:37]: Please delete the comment for the function that was removed.

File: pkg/component/component.go [hunk 2]
```diff
@@ -35,26 +35,8 @@ func GetComponentTypeFromDevfileMetadata(metadata devfile.DevfileMetadata) strin
...
-	var projectType string
-	if metadata.ProjectType != "" {
-		projectType = metadata.ProjectType
-	} else {
-		projectType = NotAvailable
-	}
-	return projectType
-}
 
 // GetLanguageFromDevfileMetadata returns component type from the devfile metadata
```
Comment (dharmit) [inline pkg/component/component.go:39]: Please delete the comment for the function that was removed.

File: pkg/devfile/image/image.go [hunk 3]
```diff
@@ -56,7 +57,7 @@ func BuildPushImages(devfileObj parser.DevfileObj, path string, push bool) error
 
 // BuildPushSpecificImage build an image defined in the devfile
 // If push is true, also push the image to its registry
-func BuildPushSpecificImage(devfileObj parser.DevfileObj, devfilePath string, component devfile.Component, push bool) error {
+func BuildPushSpecificImage(devfilePath string, component devfile.Component, push bool) error {
```
Comment (dharmit) [inline pkg/devfile/image/image.go:60]: Can you update the function's comment since we are removing `devfileObj`?

**Review by dharmit — CHANGES_REQUESTED** [review]
/approve

**Comment by valaparthvi** [comment]
/override Kubernetes Integration Tests/Kubernetes Integration Tests

**Comment by valaparthvi** [comment]
/override Unit Tests/Unit Tests
IBM test passes.

**Pull request merged**